pages/api/account/[...id].ts

```diff
@@ -0,0 +1,47 @@
+import type { NextApiResponse } from "next"
+import type { NextRequest } from "next/server"
+import { sierraClient } from "../../../src/server/sierraClient"
+import initializePatronTokenAuth from "../../../src/server/auth"
+
+/**
+ * API route handler
+ */
+
+// TODO: api/account/settings/{patronId}
+// TODO: api/account/update-pin/{patronId}
+
+export default async function handler(req: NextRequest, res: NextApiResponse) {
+  const patronTokenResponse = await initializePatronTokenAuth(req)
+  const patronId = patronTokenResponse.decodedPatron?.sub
+  if (!patronId) {
+    res.status(403).json({
+      message: "No authenticated patron",
+    })
+  }
+  const checkoutRenewalMatch = req.url.match(/\/checkouts\/(\d+)\/renewal$/)
```

Comment:
I think the comparison you want to make, according to the ticket, is `patronId` against the id from the `req.url`.

Some thoughts:
- should the file name have the catch all for the id? I think `/[id].ts` is okay and you should be able to do `req.query.id` for the variable `checkoutId`.
- Then you can compare `patronId` against the value from `checkoutId` to make sure we have the same patron.

Another approach can be to make this handler a POST request and instead of doing `/api/account/id`, you would pass the id as part of the body object. In that case you'd have a larger if statement for `if (request.method === "POST") {....` and `checkoutId` would be `req.body.id`.

Reply:
The POST approach might be more "secure" since we are not putting the patron's id in the url request. I vaguely remember talking about this but forgot where we landed @charmingduchess .

Reply:
So `patronId` is not the same thing as the `checkoutId` from the url but (after some offline discussion previously) I think it's sufficient to just check that the user is logged in before they make the request– the case that the user could be logged in and somehow viewing and renewing someone else's checkout doesn't seem possible. So the handler is really checking what you're trying to do (which will soon need to be distinguished from https://jira.nypl.org/browse/SCC-3954)